In [54]:
import pathlib
import pandas as pd
import xml.etree.ElementTree as et

# define the path
currentDirectory = pathlib.Path('XML')

# define the pattern
currentPattern = "*.xml"
all_xml_files = []
for currentFile in currentDirectory.glob(currentPattern):
    all_xml_files.append(currentFile.parts[-1])
    
print(all_xml_files)

['American_Gangster.xml', 'American_Pie.xml', 'Andaz_Apna_Apna.xml', 'Anna_Karenina.xml', 'A_Fish_Called_Wanda.xml', 'A_Goofy_Movie.xml', 'A_Million_Ways_to_Die_in_the_West.xml', 'A_Single_Man.xml', 'Babar__The_Movie.xml', 'Bhoothnath_Returns.xml', 'Big_Hero_6.xml', 'Black_Swan.xml', 'Bobby_Jasoos.xml', 'Boogie_Nights.xml', 'Bronson.xml', 'Bros_Before_Hos.xml', 'Brother_Bear.xml', 'Casino_Royale.xml', 'Chef.xml', 'Dirty_Dancing.xml', 'Donnie_Darko.xml', 'Drive.xml', 'Elsa___Fred.xml', 'Excuse_My_French.xml', 'Exodus__Gods_and_Kings.xml', 'Full_Metal_Jacket.xml', 'Grand_Piano.xml', 'Homies.xml', 'Hook.xml', 'Hotel_Transylvania.xml', 'Inherent_Vice.xml', 'Interstellar.xml', 'Into_the_Wild.xml', 'Into_the_Woods.xml', 'Invincible.xml', 'Jack_Reacher.xml', 'Jerry_Maguire.xml', 'John_Wick.xml', 'Khoobsurat.xml', 'Khumba.xml', "Let's_Be_Cops.xml", 'Le_Samourai.xml', 'Live_Nude_Girls.xml', 'Looney_Tunes__Back_in_Action.xml', 'Lord_of_the_Flies.xml', 'Maine_Pyar_Kiya.xml', 'Mean_Girls.xml', 'Mo

In [62]:
import pathlib  as sk
xml.etree.__version__

NameError: name 'xml' is not defined

In [55]:
def parse_XML(xml_file, name, df_cols):
            global out_df

            xtree = et.parse(xml_file)
            xroot = xtree.getroot()
            fin = []
            res = []
            res.append(name.split(".")[0])
            
            for node in xroot:
                for i in node_cols:
                    res.append(node.attrib.get(i))

            fin.append(res)
            print(fin)
            out_df = pd.DataFrame(data=fin, columns =meta_cols)
            return out_df
        
def parse_XML1(xml_file, name, df_cols, df):
            global out_df
            xtree = et.parse(xml_file)
            xroot = xtree.getroot()
            fin = []
            res = []
            if name.split(".")[0] == "Mr":
                res.append("Mr._Go")
            else:
                res.append(name.split(".")[0])
            for node in xroot:
                for i in node_cols:
                    res.append(node.attrib.get(i))

            fin.append(res)
            df1 = pd.DataFrame(data=fin, columns =meta_cols)
            
            frames = [df, df1]
            
            out_df = pd.concat(frames)
            
            return out_df

In [28]:
meta_cols = ["filename","language", "year", "genre", "country",
            "runtime", "rated"]

node_cols = ["language", "year", "genre", "country",
            "runtime", "rated"]

user_rating_cols = ["metascore", "imdbRating", "tomatoRating", "tomatoUserRating"]
user_and_meta_cols = ["language", "year", "genre", "country",
            "runtime", "rated", "metascore", "imdbRating", "tomatoRating", "tomatoUserRating"]
user = False
meta = True

In [29]:
if user == True:
    #meta_cols=["filename","metascore", "imdbRating", "tomatoMeter", "tomatoUserMeter"]
    meta_cols=["filename","language", "year", "genre", "country",
            "runtime", "rated"]
    node_cols=user_rating_cols
if user == True and meta == True:
    meta_cols=["filename","language", "year", "genre", "country",
            "runtime", "rated", "metascore", "imdbRating", "tomatoRating", "tomatoUserRating"]
    node_cols=user_and_meta_cols
c = 1
for i in all_xml_files:
    if c==1:
        parse_XML("XML/" + i, i,  meta_cols)
        c=43
    else:
        parse_XML1("XML/"+i, i, meta_cols, out_df)

[['American_Gangster', 'English', '2007', 'Biography, Crime, Drama', 'USA, UK', '157 min', 'R']]


In [30]:
df_ground = pd.read_excel("dev_set_groundtruth_and_trailers.xls",index_col=False)

In [31]:
a = df_ground["filename"].sort_values()
b = out_df["filename"].sort_values()

for i in a.values:
    if i in b.values:
        continue
    else:
        print(i)

In [32]:
df_join = pd.merge(df_ground, out_df, on="filename", how="inner")
#df_join = df_join.drop(["movie", "filename", "trailer"], axis=1)
df_join

movie                             filename  \
0                           Seventh Son                          Seventh_Son   
1                         Welcome to Me                        Welcome_to_Me   
2                             The Judge                            The_Judge   
3       Transformers  Age of Extinction      Transformers__Age_of_Extinction   
4                      The Normal Heart                     The_Normal_Heart   
5                 The Phantom Tollbooth                The_Phantom_Tollbooth   
6                       Andaz Apna Apna                      Andaz_Apna_Apna   
7                    Hotel Transylvania                   Hotel_Transylvania   
8                            The Matrix                           The_Matrix   
9                         Into the Wild                        Into_the_Wild   
10                        Jerry Maguire                        Jerry_Maguire   
11                    Lord of the Flies                    Lord_of_the_Flies   
12                     The Great Gatsby                     The_Great_Gatsby   
13                        Inherent Vice                        Inherent_Vice   
14                         Donnie Darko                         Donnie_Darko   
15                        Anna Karenina                        Anna_Karenina   
16                    American Gangster                    American_Gangster   
17                    Full Metal Jacket                    Full_Metal_Jacket   
18                   Words and Pictures                   Words_and_Pictures   
19                            John Wick                            John_Wick   
20               The Fault in Our Stars               The_Fault_in_Our_Stars   
21                          The Cobbler                          The_Cobbler   
22  The Hobbit  The Desolation of Smaug  The_Hobbit__The_Desolation_of_Smaug   
23                         Jack Reacher                         Jack_Reacher   
24                     The Book of Life                     The_Book_of_Life   
25                        Let's Be Cops                        Let's_Be_Cops   
26                         A Single Man                         A_Single_Man   
27                        Casino Royale                        Casino_Royale   
28                     Ode to My Father                     Ode_to_My_Father   
29                        Dirty Dancing                        Dirty_Dancing   
..                                  ...                                  ...   
65                     Some Like It Hot                     Some_Like_It_Hot   
66                           The Reader                           The_Reader   
67                         The Birdcage                         The_Birdcage   
68             The Theory of Everything             The_Theory_of_Everything   
69                           Black Swan                           Black_Swan   
70                      Maine Pyar Kiya                      Maine_Pyar_Kiya   
71                          Still Alice                          Still_Alice   
72                 My Week with Marilyn                 My_Week_with_Marilyn   
73                   Bhoothnath Returns                   Bhoothnath_Returns   
74                          Grand Piano                          Grand_Piano   
75                       Into the Woods                       Into_the_Woods   
76              The Blair Witch Project              The_Blair_Witch_Project   
77                  Monsters University                  Monsters_University   
78                               Mr. Go                               Mr._Go   
79            This Is Where I Leave You            This_Is_Where_I_Leave_You   
80                    The Bourne Legacy                    The_Bourne_Legacy   
81                           Wild Tales                           Wild_Tales   
82                      The Dark Knight                      The_Dark_Knight   
83                       Predestin

In [33]:
import numpy as np
if user == False or meta == True:
    count=0
    for i in df_join["genre"].values:
        df_join["genre"].values[count] = i.replace(" ", "")
        count+=1

    count=0
    for i in df_join["language"].values:
        df_join["language"].values[count] = i.replace(" ", "")
        count+=1


    count=0
    for i in df_join["country"].values:
        df_join["country"].values[count] = i.replace(" ", "")
        count+=1
        
    genre = df_join.genre.str.split(",", expand=True).stack()
    language = df_join.language.str.split(",", expand=True).stack()
    country = df_join.country.str.split(",", expand=True).stack()
    
    df_join = pd.concat([df_join, 
                pd.get_dummies(genre, prefix="g").groupby(level=0).sum()],axis=1) \
                .drop(["genre"],axis=1)

    df_join = pd.concat([df_join, 
                    pd.get_dummies(language, prefix="l").groupby(level=0).sum()],axis=1) \
                    .drop(["language"],axis=1)

    df_join = pd.concat([df_join, 
                    pd.get_dummies(country, prefix="c").groupby(level=0).sum()],axis=1) \
                    .drop(["country"],axis=1)

In [34]:
df_join

movie                             filename  \
0                           Seventh Son                          Seventh_Son   
1                         Welcome to Me                        Welcome_to_Me   
2                             The Judge                            The_Judge   
3       Transformers  Age of Extinction      Transformers__Age_of_Extinction   
4                      The Normal Heart                     The_Normal_Heart   
5                 The Phantom Tollbooth                The_Phantom_Tollbooth   
6                       Andaz Apna Apna                      Andaz_Apna_Apna   
7                    Hotel Transylvania                   Hotel_Transylvania   
8                            The Matrix                           The_Matrix   
9                         Into the Wild                        Into_the_Wild   
10                        Jerry Maguire                        Jerry_Maguire   
11                    Lord of the Flies                    Lord_of_the_Flies   
12                     The Great Gatsby                     The_Great_Gatsby   
13                        Inherent Vice                        Inherent_Vice   
14                         Donnie Darko                         Donnie_Darko   
15                        Anna Karenina                        Anna_Karenina   
16                    American Gangster                    American_Gangster   
17                    Full Metal Jacket                    Full_Metal_Jacket   
18                   Words and Pictures                   Words_and_Pictures   
19                            John Wick                            John_Wick   
20               The Fault in Our Stars               The_Fault_in_Our_Stars   
21                          The Cobbler                          The_Cobbler   
22  The Hobbit  The Desolation of Smaug  The_Hobbit__The_Desolation_of_Smaug   
23                         Jack Reacher                         Jack_Reacher   
24                     The Book of Life                     The_Book_of_Life   
25                        Let's Be Cops                        Let's_Be_Cops   
26                         A Single Man                         A_Single_Man   
27                        Casino Royale                        Casino_Royale   
28                     Ode to My Father                     Ode_to_My_Father   
29                        Dirty Dancing                        Dirty_Dancing   
..                                  ...                                  ...   
65                     Some Like It Hot                     Some_Like_It_Hot   
66                           The Reader                           The_Reader   
67                         The Birdcage                         The_Birdcage   
68             The Theory of Everything             The_Theory_of_Everything   
69                           Black Swan                           Black_Swan   
70                      Maine Pyar Kiya                      Maine_Pyar_Kiya   
71                          Still Alice                          Still_Alice   
72                 My Week with Marilyn                 My_Week_with_Marilyn   
73                   Bhoothnath Returns                   Bhoothnath_Returns   
74                          Grand Piano                          Grand_Piano   
75                       Into the Woods                       Into_the_Woods   
76              The Blair Witch Project              The_Blair_Witch_Project   
77                  Monsters University                  Monsters_University   
78                               Mr. Go                               Mr._Go   
79            This Is Where I Leave You            This_Is_Where_I_Leave_You   
80                    The Bourne Legacy                    The_Bourne_Legacy   
81                           Wild Tales                           Wild_Tales   
82                      The Dark Knight                      The_Dark_Knight   
83                       Predestin

In [35]:
if user == True:
    df_join.replace("N/A", np.nan, inplace=True)


In [36]:
df_join.dtypes

df_join = df_join.sort_values(by=["filename"])

df_join = df_join.drop(["movie", "filename","trailer"], axis=1)
df_join

goodforairplane  year  runtime     rated  g_Action  g_Adventure  \
46                1  1988  108 min         R         0            0   
93                0  1995   78 min         G         0            1   
85                1  2014  116 min         R         0            0   
26                1  2009   99 min         R         0            0   
16                1  2007  157 min         R         0            0   
37                1  1999   95 min         R         0            0   
6                 1  1994  160 min        PG         0            0   
15                1  2012  129 min         R         0            0   
32                0  1989   70 min         G         0            1   
73                0  2014  155 min       N/A         0            0   
58                1  2014  102 min        PG         1            1   
69                1  2010  108 min         R         0            0   
39                1  2014  121 min       N/A         0            0   
40                0  1997  155 min         R         0            0   
87                0  2008   92 min         R         1            0   
34                1  2013   87 min       N/A         0            0   
31                0  2003   85 min         G         0            1   
27                1  2006  144 min     PG-13         1            1   
42                0  2014  114 min         R         0            0   
29                1  1987  100 min     PG-13         0            0   
14                1  2001  113 min         R         0            0   
86                0  2011  100 min         R         0            0   
51                0  2005  108 min        PG         0            0   
41                0  2014   99 min       N/A         0            0   
63                0  2014  150 min     PG-13         1            1   
17                1  1987  116 min         R         0            0   
74                1  2013   90 min         R         0            0   
47                1  2015  100 min       N/A         0            0   
62                0  1991  142 min        PG         0            1   
7                 1  2012   91 min        PG         0            0   
..              ...   ...      ...       ...       ...          ...   
67                1  1996  117 min         R         0            0   
76                1  1999   81 min         R         0            0   
24                1  2014   95 min        PG         0            1   
80                0  2012  135 min     PG-13         1            1   
21                0  2014   99 min     PG-13         0            0   
82                0  2008  152 min     PG-13         1            0   
20                0  2014  126 min     PG-13         0            0   
12                0  2013  143 min     PG-13         0            0   
22                1  2013  161 min     PG-13         0            1   
56                1  2012  142 min     PG-13         0            1   
61                0  2013  146 min     PG-13         0            1   
59                1  2014  114 min     PG-13         0            0   
2                 0  2014  141 min         R         0            0   
88                0  2013   98 min         R         0            0   
44                1  2003   94 min        PG         0            1   
8                 1  1999  136 min         R         1            0   
49                1  2014  113 min     PG-13         1            0   
4                 1  2014  132 min     TV-MA         0            0   
33                0  1998  127 min        PG         0            1   
5                 1  1970   90 min         G         0            1   
66                1  2008  124 min         R         0            0   
89                0  1941   74 min  APPROVED         0            0   
68                1  2014  123 min     PG-13         0            0   
79                1  2014  103 min         R         0            0   
36                1  1990  1

In [37]:
df_join.columns
df_join = df_join.astype("category")

In [39]:
df_join.to_csv("meta_sorted.csv",index=False)

=== Detailed Accuracy By Class === user ratings

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,233    0,346    0,357      0,233    0,282      -0,124   0,413     0,400     '(-inf-0.5]'
                 0,654    0,767    0,507      0,654    0,571      -0,124   0,416     0,489     '(0.5-inf)'
Weighted Avg.    0,463    0,577    0,439      0,463    0,440      -0,124   0,415     0,449     



=== Detailed Accuracy By Class === meta+rating

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,116    0,077    0,556      0,116    0,192      0,067    0,522     0,489     '(-inf-0.5]'
                 0,923    0,884    0,558      0,923    0,696      0,067    0,522     0,558     '(0.5-inf)'
Weighted Avg.    0,558    0,519    0,557      0,558    0,468      0,067    0,522     0,526     


In [42]:
## VISUAL 
# define the path
currentDirectory = pathlib.Path('D:/TU/Exp-Design-DS/CoE_dataset/Test_Set/vis_descriptors')

# define the pattern
currentPattern = "*.csv"
all_visual_files = []
for currentFile in currentDirectory.glob(currentPattern):
    all_visual_files.append(currentFile.parts[-1])
    
print(all_visual_files)

['10.000_Km.csv', '12_Years_a_Slave.csv', '21_Jump_Street.csv', '2_States.csv', 'Aanmodderfakker.csv', 'Alexander.csv', 'Alexander_and_the_Terrible,_Horrible,_No_Good,_Very_Bad_Day.csv', 'Alice_in_Wonderland.csv', 'American_Sniper.csv', 'Anastasia.csv', 'And_So_It_Goes.csv', 'Antz.csv', 'Avatar.csv', "A_Bug's_Life.csv", 'A_Little_Chaos.csv', 'A_Most_Wanted_Man.csv', 'A_Simple_Life.csv', 'Bad_News_Bears.csv', 'Banklady.csv', 'Beetlejuice.csv', 'Begin_Again.csv', 'Belle_de_Jour.csv', 'Bewakoofiyaan.csv', 'Bhoothnath.csv', 'Big_Game.csv', 'Billy_Elliot.csv', 'Birdman__Or_(The_Unexpected_Virtue_of_Ignorance).csv', 'Blended.csv', 'Blow_Out.csv', 'Bride_Flight.csv', 'Bridges_of_Sarajevo.csv', 'Burlesque.csv', 'Captain_America__The_Winter_Soldier.csv', 'Cascadeur.csv', 'Charlie_and_the_Chocolate_Factory.csv', 'Clash_of_the_Titans.csv', 'Contact.csv', 'Dances_with_Wolves.csv', 'Dilwale_Dulhania_Le_Jayenge.csv', 'Divergent.csv', 'Django_Unchained.csv', 'Dolphin_Tale_2.csv', 'Dorsvloer_vol_confe

In [45]:
df = pd.read_csv("D:/TU/Exp-Design-DS/CoE_dataset/Test_Set/vis_descriptors/" + all_visual_files[0], header=None)
df

0    1    2    3    4    5    6    7    8    9      ...          816  \
0    0    0    0    0    0    0    0    0    0    0    ...     518400.0   
1    0    0    0    0    0    0    0    0    0    0    ...     518400.0   

        817           818       819       820       821        822       823  \
0  269700.0  6.103500e-07  0.001645  0.000002  0.001645  1638400.0  269700.0   
1  269700.0  6.103500e-07  0.001645  0.000002  0.001645  1638400.0  269700.0   

        824       825  
0  518400.0  269700.0  
1  518400.0  269700.0  

[2 rows x 826 columns]

In [47]:
c = 1

for i in all_visual_files:
    if c==1:
        df_vis = pd.read_csv("D:/TU/Exp-Design-DS/CoE_dataset/Test_Set/vis_descriptors/" + i, header=None)
        all_features = []
        for index, row in df_vis.iterrows():
            for j in row.values:
                all_features.append(j)
        df_vis_concat = pd.DataFrame([all_features], columns=list(range(0, 1652)))
        name = i[:-4]
        df_vis_concat.insert(0, "filename", name)
        first_row = df_vis_concat.values.tolist()
        c=43
    else:
        df_vis = pd.read_csv("D:/TU/Exp-Design-DS/CoE_dataset/Test_Set/vis_descriptors/" + i, header=None)
        all_features = []
        for index, row in df_vis.iterrows():
            for j in row.values:
                all_features.append(j)
        df_temp = pd.DataFrame([all_features], columns=list(range(0, 1652)))
        name = i[:-4]
        df_temp.insert(0, "filename", name)
        first_row.append(df_temp.values.tolist()[0])
df_vis_fin = pd.DataFrame(first_row)
df_vis_fin

0         1         2     \
0                                            10.000_Km  0.000000  0.000000   
1                                     12_Years_a_Slave  0.000000  0.193000   
2                                       21_Jump_Street  0.000000  0.000000   
3                                             2_States  0.033181  0.089760   
4                                      Aanmodderfakker  0.271970  0.334210   
5                                            Alexander  0.000000  0.000000   
6    Alexander_and_the_Terrible,_Horrible,_No_Good,...  0.000000  0.000000   
7                                  Alice_in_Wonderland  0.000000  0.000000   
8                                      American_Sniper  0.000000  0.000000   
9                                            Anastasia  0.000000  0.000000   
10                                      And_So_It_Goes  0.000000  0.000000   
11                                                Antz  0.434330  0.447210   
12                                              Avatar  0.134080  0.203490   
13                                        A_Bug's_Life  0.000000  0.000000   
14                                      A_Little_Chaos  0.272800  0.260960   
15                                   A_Most_Wanted_Man  0.000000  0.000000   
16                                       A_Simple_Life  0.000000  0.000000   
17                                      Bad_News_Bears  0.000000  0.000000   
18                                            Banklady  0.053247  0.145600   
19                                         Beetlejuice  0.000000  0.000000   
20                                         Begin_Again  0.290990  0.326400   
21                                       Belle_de_Jour  0.005963  0.023951   
22                                       Bewakoofiyaan  0.000000  0.000000   
23                                          Bhoothnath  0.363190  0.409210   
24                                            Big_Game  0.000000  0.000000   
25                                        Billy_Elliot  0.002460  0.003696   
26    Birdman__Or_(The_Unexpected_Virtue_of_Ignorance)  0.000000  0.000000   
27                                             Blended  0.000000  0.000000   
28                                            Blow_Out  0.000000  0.000000   
29                                        Bride_Flight  0.209470  0.246400   
..                                                 ...       ...       ...   
193                                 The_Simpsons_Movie  0.400190  0.264370   
194                                 The_Social_Network  0.000000  0.000000   
195                            The_Talented_Mr._Ripley  0.000000  0.000000   
196                                  The_Trip_to_Italy  0.000000  0.000000   
197                                         The_Voices  0.297000  0.320510   
198                                  The_Water_Diviner  0.280960  0.333330   
199            The_White_Haired_Witch_of_Lunar_Kingdom  0.000000  0.266250   
200                            The_Wolf_of_Wall_Street  0.000000  0.000000   
201                               Thor__The_Dark_World  0.000000  0.000000   
202                                            Titanic  0.247200  0.268550   
203                                          Tommy_Boy  0.000000  0.000000   
204                                      Trainspotting  0.000000  0.000000   
205                                              Trash  0.238580  0.249240   
206                                           Unbroken  0.287890  0.320650   
207                                           Valkyrie  0.000000  0.000000   
208                                  Verliefd_op_Ibiza  0.137650  0.276900   
209                                      Walk_the_Line  0.000000  0.000000   
210                                            Warrior  0.275500  0.286510   
211                                     Wassup_Rockers  0.000000  0.000000   
212                                           Watchmen  0.000000  0.0000

In [48]:


z = list(range(0,1652))
z.insert(0, "filename")
colnames = z
df_vis_fin.columns = colnames
df_vis_fin

filename         0         1  \
0                                            10.000_Km  0.000000  0.000000   
1                                     12_Years_a_Slave  0.000000  0.193000   
2                                       21_Jump_Street  0.000000  0.000000   
3                                             2_States  0.033181  0.089760   
4                                      Aanmodderfakker  0.271970  0.334210   
5                                            Alexander  0.000000  0.000000   
6    Alexander_and_the_Terrible,_Horrible,_No_Good,...  0.000000  0.000000   
7                                  Alice_in_Wonderland  0.000000  0.000000   
8                                      American_Sniper  0.000000  0.000000   
9                                            Anastasia  0.000000  0.000000   
10                                      And_So_It_Goes  0.000000  0.000000   
11                                                Antz  0.434330  0.447210   
12                                              Avatar  0.134080  0.203490   
13                                        A_Bug's_Life  0.000000  0.000000   
14                                      A_Little_Chaos  0.272800  0.260960   
15                                   A_Most_Wanted_Man  0.000000  0.000000   
16                                       A_Simple_Life  0.000000  0.000000   
17                                      Bad_News_Bears  0.000000  0.000000   
18                                            Banklady  0.053247  0.145600   
19                                         Beetlejuice  0.000000  0.000000   
20                                         Begin_Again  0.290990  0.326400   
21                                       Belle_de_Jour  0.005963  0.023951   
22                                       Bewakoofiyaan  0.000000  0.000000   
23                                          Bhoothnath  0.363190  0.409210   
24                                            Big_Game  0.000000  0.000000   
25                                        Billy_Elliot  0.002460  0.003696   
26    Birdman__Or_(The_Unexpected_Virtue_of_Ignorance)  0.000000  0.000000   
27                                             Blended  0.000000  0.000000   
28                                            Blow_Out  0.000000  0.000000   
29                                        Bride_Flight  0.209470  0.246400   
..                                                 ...       ...       ...   
193                                 The_Simpsons_Movie  0.400190  0.264370   
194                                 The_Social_Network  0.000000  0.000000   
195                            The_Talented_Mr._Ripley  0.000000  0.000000   
196                                  The_Trip_to_Italy  0.000000  0.000000   
197                                         The_Voices  0.297000  0.320510   
198                                  The_Water_Diviner  0.280960  0.333330   
199            The_White_Haired_Witch_of_Lunar_Kingdom  0.000000  0.266250   
200                            The_Wolf_of_Wall_Street  0.000000  0.000000   
201                               Thor__The_Dark_World  0.000000  0.000000   
202                                            Titanic  0.247200  0.268550   
203                                          Tommy_Boy  0.000000  0.000000   
204                                      Trainspotting  0.000000  0.000000   
205                                              Trash  0.238580  0.249240   
206                                           Unbroken  0.287890  0.320650   
207                                           Valkyrie  0.000000  0.000000   
208                                  Verliefd_op_Ibiza  0.137650  0.276900   
209                                      Walk_the_Line  0.000000  0.000000   
210                                            Warrior  0.275500  0.286510   
211                                     Wassup_Rockers  0.000000  0.000000   
212                                           Watchmen  0.000000  0.

In [53]:
df_ground = pd.read_csv("D:/TU/Exp-Design-DS/CoE_dataset/Test_Set/test_labels_sorted_cleaned.csv") 

df_join1 = pd.merge(df_ground, df_vis_fin, on="filename", how="inner")
df_join1 = df_join1.drop(["filename"], axis=1)
df_join1 = df_join1.sort_values(by=["movie"])

df_join1  = df_join1.drop(["movie"], axis=1)

df_join1.to_csv("visual_test_sorted.csv",index=False)


In [183]:
df_join.to_csv("meta+visual.csv",index=False)

=== Detailed Accuracy By Class === visual

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,163    0,154    0,467      0,163    0,241      0,012    0,485     0,445     '(-inf-0.5]'
                 0,846    0,837    0,550      0,846    0,667      0,012    0,485     0,543     '(0.5-inf)'
Weighted Avg.    0,537    0,528    0,512      0,537    0,474      0,012    0,485     0,499 


=== Detailed Accuracy By Class === visual+meta

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,233    0,231    0,455      0,233    0,308      0,002    0,482     0,442     '(-inf-0.5]'
                 0,769    0,767    0,548      0,769    0,640      0,002    0,482     0,536     '(0.5-inf)'
Weighted Avg.    0,526    0,525    0,506      0,526    0,490      0,002    0,482     0,493 